src/System.Private.CoreLib/shared/System/Runtime/InteropServices/ArrayWithOffset.cs

```diff
@@ -16,7 +26,30 @@ public ArrayWithOffset(object array, int offset)
             m_array = array;
             m_offset = offset;
             m_count = 0;
```

Comment:
The field is initialized twice and setting other ones after arguments checks might be better too